File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;
class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int [score.length];
        
        PriorityQueue <Integer> pq = new PriorityQueue <>();
        int day = 1 ;
        int idx = 0;
        for (int s : score)  {
            if (day > k) {
                int cur_min = pq.peek();
                if (cur_min  < s) {
                    pq.poll();
                    pq.offer(s);
                }
            }else  {
                pq.offer(s);    
            }
            
            
            day +=1; 
            answer[idx] = pq.peek();
            idx+=1;
            
            
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq
def solution(k, score):
    answer = []
    temp = []
    
    day = 1
    
    for current_score in score :
        if day > k : 
            if len(temp) == k : 
                cur_min_val = temp[0]
                if cur_min_val < current_score :
                    heapq.heappush (temp, current_score)
                    heapq.heappop(temp)
                
                    
        else: 
            heapq.heappush(temp, current_score)   
        
        day +=1
        answer.append(temp[0])
    
    
    return answer